src/Symfony/Component/Security/Core/Authorization/Voter/AbstractVoter.php

```diff
@@ -65,6 +65,12 @@ public function vote(TokenInterface $token, $object, array $attributes)
         // abstain vote by default in case none of the attributes are supported
         $vote = self::ACCESS_ABSTAIN;
 
+        $reflector = new \ReflectionMethod($this, 'voteOnAttribute');
+        $isNewOverwritten = $reflector->getDeclaringClass()->getName() !== 'Symfony\Component\Security\Core\Authorization\Voter\AbstractVoter';
```

Comment:
I would use ``__CLASS__`` here rather than the class name in a string. It makes it clearer than it is the current class, and is less likely to break if we rename things